networkit/gephi/pyclient.py

```diff
@@ -33,85 +33,220 @@
-        if self.unflushedDumps > self.autoflush:
-            self.flush()
-            self.unflushedDumps = 0
-    def _send(self, data):
-        print('passing')
-        pass
-
-    def add_node(self, id, flush=True, **attributes):
-        self.data += json.dumps(self.peh({"an":{id:attributes}})) + '\r\n'
-        self.incrementUnflushedDumps()
-
-    def change_node(self, id, flush=True, **attributes):
-        self.data += json.dumps(self.peh({"cn":{id:attributes}})) + '\r\n'
-        self.incrementUnflushedDumps()
-
-    def delete_node(self, id):
-        self._send(json.dumps(self.peh({"dn":{id:{}}})) + '\r\n')
-
-    def add_edge(self, id, source, target, directed=True, **attributes):
-        attributes['source'] = source
-        attributes['target'] = target
-        attributes['directed'] = directed
-        self.data += json.dumps(self.peh({"ae":{id:attributes}})) + '\r\n'
-        self.incrementUnflushedDumps()
-
-    def change_edge(self, id, source, target, directed=True, **attributes):
-        attributes['source'] = source
-        attributes['target'] = target
-        attributes['directed'] = directed
-        self.data += json.dumps(self.peh({"ce":{id:attributes}})) + '\r\n'
-        self.incrementUnflushedDumps()
-
-    def delete_edge(self, id):
-        self._send(json.dumps(self.peh({"de":{id:{}}})) + '\r\n')
-
-    def clean(self):
-        self._send(json.dumps(self.peh({"dn":{"filter":"ALL"}})) + '\r\n')
+	"""
+	JSONClient(autoflush=0, enable_timestamps=False, process_event_hook=None)
+
+	Supporter class for Gephiclient. Used to encapsulate data in JSON-format.
+
+	Parameters
+	----------
+	autoflush : int, optional
+		Enables automatic after certain data additions. Default = 0
+	enable_timestamps : bool, optional
+		Enable timestamps for data entries. Default: False
+	process_event_hook : function, optional
+		Custom event hook function. Default: None
+	"""
+	def __init__(self, autoflush=0, enable_timestamps=False, process_event_hook=None):
+		self.data = ""
+		self.autoflush = autoflush
+		self.unflushedDumps = 0
+		self.enable_timestamps = enable_timestamps
+
+		if enable_timestamps:
+			def default_peh(event):
+				event['t'] = int(time.time())
+				return event
+		else:
+			default_peh = lambda e: e
+		if process_event_hook is None:
+			self.peh = default_peh
+		else:
+			self.peh = lambda e: default_peh(process_event_hook(e))
+
+	def flush(self):
+		"""
+		flush()
+
+		Sends all data and clear buffer.
+		"""
+		if len(self.data) > 0:
+			self._send(self.data)
+			self.data = ""
+		
+	def incrementUnflushedDumps(self):
+		"""
+		incrementUnflushedDumps()
+
+		Increment number of unflushed dumps. Is resetted based on the value of autoflush.
+		"""
+		self.unflushedDumps = self.unflushedDumps + 1
+		if self.unflushedDumps > self.autoflush:
+			self.flush()
+			self.unflushedDumps = 0
+	def _send(self, data):
+		print('passing')
+		pass
+
+	def add_node(self, id, flush=True, **attributes):
```

Comment:
These function names could also be changed to camelCase

Reply:
Right, should I deprecate the previous ones (since it is only a helper class)?

Reply:
Yes, but better to do that in a separate PR